### DISCUSSION TITLE
Prismic CMS on Firebase

### DISCUSSION THREAD

**Discussion by ijsamuel** [vercel/next.js#39345] [Help]
I am developing my site on Prismic CMS. I also have an existing site hosted on Firebase - just static web pages. How do I deploy my Prismic Next.JS application to Firebase? Firebase needs a "public" folder but there is no public folder for Prismic. Appreciate any help, thanks!

**Comment by Babbili** [comment]
HI @ijsamuel one of the ways to deploy Next Js app to Firebase is to to do a [Static HTML Export](https://nextjs.org/docs/advanced-features/static-html-export) of your Next Js app,
This will allows you to export your Next.js application to static HTML, and it can be done by updating the **build script** in `package.json` file to use `next export`
```json
//  package.json

"scripts": {
  "build": "next build && next export"   // add 'next export' to the build script
}
```
Now on running `npm run build` it will generate an `out` directory and build an HTML version of your app in it.
Update `firebase.json` file to set the `out` directory as public
```json
// firebase.json

{
  "hosting": {
    "public": "out",  // 👈
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ],
    "rewrites": [
      {
        "source": "**",
        "destination": "/index.html"
      }
    ]
  }
}
```
Now you're ready to go `firebase deploy`
Read the [Unsupported Features in Static HTML export](https://nextjs.org/docs/advanced-features/static-html-export#unsupported-features) to avoid using them.

**Reply by ijsamuel** [reply]
thanks @Babbili - I added next export. When I go to npm run build. I get below error. I have installed node-fetch. And then added 
`import fetch from "node-fetch";` to `prismicio.js` and then in the same updated createClient to:
export const createClient = (config = {}) => {
  const client = prismic.createClient(sm.apiEndpoint, {
    fetch,
    routes: [
      { type: "page", path: "/:uid" },
      { type: "settings", path: "/" },
      { type: "navigation", path: "/" },
    ],
  });

but still getting same error. Below is the error:


PrismicError: A valid fetch implementation was not provided. In environments where fetch is not available (including Node.js), a fetch implementation must be provided via a polyfill or the `fetch` option.
    at new Client (C:\temp\MeCoding\FireBase\Prismic\pows\node_modules\@prismicio\client\dist\index.cjs:280:13)
    at Object.createClient (C:\temp\MeCoding\FireBase\Prismic\pows\node_modules\@prismicio\client\dist\index.cjs:252:61)
    at nextConfig (C:\temp\MeCoding\FireBase\Prismic\pows\next.config.js:7:26)
    at Object.normalizeConfig (C:\temp\MeCoding\FireBase\Prismic\pows\node_modules\next\dist\server\config-shared.js:109:18)
    at Object.loadConfig [as default] (C:\temp\MeCoding\FireBase\Prismic\pows\node_modules\next\dist\server\config.js:74:53)
    at async Span.traceAsyncFn (C:\temp\MeCoding\FireBase\Prismic\pows\node_modules\next\dist\trace\trace.js:79:20)
    at async C:\temp\MeCoding\FireBase\Prismic\pows\node_modules\next\dist\export\index.js:36:45
    at async Span.traceAsyncFn (C:\temp\MeCoding\FireBase\Prismic\pows\node_modules\next\dist\trace\trace.js:79:20) {
  url: undefined,
  response: undefined
}
